### ISSUE TITLE
Scheduled runs of "Deploy Nightly Demo" and "Deploy Demo Version Index" are failing

### ISSUE THREAD

**Issue by joeyparrish** [shaka-project/shaka-player#6318]
Scheduled runs of "Deploy Nightly Demo" and "Deploy Demo Version Index" are failing since upgrading actions.  Re-running the failed jobs manually always works.

This may point to a configuration issue with the default token used.  Perhaps re-running the job uses the maintainer's access instead of the bot's?  Needs investigation.

**Comment by joeyparrish** [comment]
The error is:

> Error: google-github-actions/deploy-appengine failed with: failed to execute gcloud command ...
> ERROR: (gcloud.app.deploy) There was a problem refreshing your current auth tokens:
> ('invalid_grant: Invalid JWT Signature.', ***'error': 'invalid_grant', 'error_description': 'Invalid JWT Signature.'***)

It's unclear why that would be different for different triggers.  The credentials are stored in the repo's secrets.  The action `google-github-actions/auth@v2` reads those and sets up some environment variables and local files for use later by `google-github-actions/deploy-appengine@v2`:

```yaml
      - uses: google-github-actions/auth@v2
        with:
          credentials_json: '${{ secrets.APPENGINE_DEPLOY_KEY }}'
```

We don't use `GITHUB_TOKEN` here, so there's no reason for triggering by a different user to change any of this.  And if the schedule trigger were unable to read the secret key, I would expect that the workflow would simply fail at the `auth` step.

Maybe I should confirm this.

**Comment by joeyparrish** [comment]
I looked up our deployment keys in Google Cloud, and found that all were expired.  It's not clear how this was able to work on a rerun if that were the case... but I updated the secret key and ran both workflows manually again.  They worked, but so did the manual re-runs before.

I'll wait to see if the next scheduled runs complete successfully.  If so, the updated key fixed it, and I'll close the issue.
